server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRows.java

```diff
@@ -0,0 +1,204 @@
+     */
+    @Override
+    public List<DialectType> getSupportsDialectTypes() {
+        return Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL);
+    }
+
+    /**
+     * MySQL execute 'explain' statement
+     *
+     * @param explainSql target sql
+     * @param jdbc jdbc Object
+     * @return affected rows
+     */
+    private long getMySqlAffectedRows(String explainSql, JdbcOperations jdbc) {
+
+        try {
+            List<Long> resultSet = jdbc.query(explainSql,
+                    (rs, rowNum) -> rs.getLong("rows"));
+
+            Long firstNonNullResult = resultSet.stream()
+                    .filter(Objects::nonNull)
+                    .findFirst()
+                    .orElse(null);
+
+            return firstNonNullResult != null ? firstNonNullResult : 0;
+
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to execute sql: " + explainSql);
+        }
+    }
+
+    /**
+     * OBMySQL execute 'explain' statement
+     *
+     * @param explainSql target sql
+     * @param jdbc jdbc Object
+     * @return affected rows
+     */
+    private long getOBMySqlAffectedRows(String explainSql, JdbcOperations jdbc) {
+
+        /**
+         * <pre>
+         *
+         *     explain result (json):
+         *
+         *     ==================================================    --rowNum = 1
+         *     |ID|OPERATOR          |NAME|EST.ROWS|EST.TIME(us)|    --rowNum = 2
+         *     0 |DISTRIBUTED UPDATE|    |1       |37          |     --rowNum = 3
+         *     1 |└─TABLE GET       |user|1       |5           |     --rowNum = 4
+         *     ==================================================    --rowNum = 5
+         *     ...
+         *
+         * </pre>
+         */
+        try {
+            AtomicBoolean ifFindAffectedRow = new AtomicBoolean(false);
+            List<String> queryResults = jdbc.query(explainSql, (rs, rowNum) -> rs.getString("Query Plan"));
+            List<Long> resultSet = new ArrayList<>();
+            for (int rowNum = 0; rowNum < queryResults.size(); rowNum++) {
+                String resultRow = queryResults.get(rowNum);
+                if (!ifFindAffectedRow.get() && rowNum > 2) {
+                    // Find the first non-null value in the column 'EST.ROWS'
+                    long estRowsValue = getEstRowsValue(resultRow);
+                    if (estRowsValue != 0) {
+                        ifFindAffectedRow.set(true);
+                        resultSet.add(estRowsValue);
+                    }
+                }
+                resultSet.add(null);
+            }
+
+            Long firstNonNullResult = resultSet.stream()
+                    .filter(Objects::nonNull)
+                    .findFirst()
+                    .orElse(null);
+
+            return firstNonNullResult != null ? firstNonNullResult : 0;
+
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to execute sql: " + explainSql);
```

Comment:
how to handle this exception?

Reply:
I forgot to restore it when running the unit test. now finished.